lib/phoenix/code_reloader/server.ex

```diff
@@ -13,8 +13,16 @@ defmodule Phoenix.CodeReloader.Server do
     GenServer.call(__MODULE__, :check_symlinks, :infinity)
   end
 
-  def reload!(endpoint) do
-    GenServer.call(__MODULE__, {:reload!, endpoint}, :infinity)
+  def reload!(conn, callback) do
```

Comment:
Hrm, I don't like that we went from a process that works only on the endpoint to a process that expects a connection to work. The ConnRegistry feels unnecessary, as it feels an indirection around the proxy.

Here is my suggestion: because we only run one code_reloader/server.ex, we can also start the proxy.ex when we start the code server, in the same supervision tree. Now that the proxy is always running, it will keep a list of pids it should proxy to. Then the compiler will do the following:

```elixir
original_gl = Process.group_leader
Proxy.capture(original_gl)
try do
  {..., Proxy.flush()}
after
  Proxy.uncapture()
end
```

Then, to redirect the processing to the browser, you will simply do:

```elixir
Proxy.forward_to(self())
```

Incorporating the registry functionality directly into the proxy. Then the proxy will forward to those processes when necessary and the CodeReloader will use `receive` to use those messages to render, instead of relying on callbacks.

This would allow us for example to even have routes such as `/__compile__` where you could be connected to the proxy all the time, regardless if you are also triggering the code reloader.

Does it make sense?